cpp/src_prims/distance/cosine.cuh

```diff
@@ -50,19 +202,103 @@ void cosineAlgo1(Index_ m, Index_ n, Index_ k, const InType *pA,
                  const InType *pB, OutType *pD, AccType *workspace,
                  size_t worksize, FinalLambda fin_op, cudaStream_t stream,
                  bool isRowMajor) {
-  typedef ExpandedDistanceFragmentMultiplyAdd<CosFusedDistance>
-    FragmentMultiplyAdd_;
   auto norm_op = [] __device__(AccType in) { return raft::mySqrt(in); };
 
   // Wrap fin_op to allow computing 1 - pA before calling fin_op
   auto wrapped_fin_op = [fin_op] __device__(AccType d_val, Index_ g_d_idx) {
     return fin_op(static_cast<AccType>(1.0) - d_val, g_d_idx);
   };
 
-  distanceAlgo1<InType, AccType, OutType, OutputTile_, FragmentMultiplyAdd_,
-                decltype(wrapped_fin_op), decltype(norm_op), Index_>(
-    m, n, k, pA, pB, pD, false, workspace, worksize, wrapped_fin_op, norm_op,
-    stream, isRowMajor);
+  typedef std::is_same<OutType, bool> is_bool;
+
+  if (((pA != pB) && (worksize < (m + n) * sizeof(AccType))) ||
+      (worksize < m * sizeof(AccType))) {
+    THROW("workspace size error");
+  }
```

Comment:
nitpick: you can use the `ASSERT` macro instead of explicitly writing `if` conditional followed by a `THROW`.

Reply:
ok